### ISSUE TITLE
Report an error like TS2713 for enum members and provide a corresponding code fix

### ISSUE THREAD

**Issue by amcasey** [microsoft/TypeScript#20358]
```ts
    enum Color {
        Red,
        Green,
        Blue
    }

    let a1: Color.Red.toString; // TS2694: The Namespace 'Color' has no exported member 'Red'
    let a2: Color.Red["toString"];
```

The type of `a1` is invalid - it should be like the type of `a2`.  It would be nice if the error indicated as much (which would also allow us to offer a code fix).

**Comment by amcasey** [comment]
FYI @DanielRosenwasser

**Comment by remojansen** [comment]
Hi, I would like to try to work on this issue. The error `TS2713` reads:

> Cannot access '{0}.{1}' because '{0}' is a type, but not a namespace. Did you mean to retrieve the type of the property '{1}' in '{0}' with '{0}[\"{1}\"]'?

How should the new error look like?

**Comment by mhegazy** [comment]
//cc @DanielRosenwasser for feedback here.

**Comment by ayip10** [comment]
this issue is being looked at for ghcOSD!

**Comment by yehee** [comment]
@ayip10 Hi! I was looking into the issue just now and wondering if this has already been worked on. Thanks!

**Comment by abuob** [comment]
Currently going through "Good first issue"-labels and stumbled over this one; I think this issue can be closed, it seems outdated?
The error is currently reported like this (not anymore as in the initial description):
```
enum Color {
    Red,
    Green,
    Blue
}

let a1: Color.Red.toString; // 'Color.Red.toString' refers to a value, but is being used as a type here. Did you mean 'typeof Color.Red.toString'? (2749)
let a2: Color.Red["toString"];
```
[Playground Link](https://www.typescriptlang.org/play?#code/KYOwrgtgBAwg9gGzgJygbwFBW1ASsAEwBoscBxZYUEnKAIQTGAwF8MMFgAXKAQwEYAXLEQoAdPgJiucAMpdkASxABzANxQA9JqgByeEmQTC0uQuUrdUSgDNgyAM5QZfKADdejYESgAjMDyKTr7AFlBgDoR8TrzOAJ4ADsBQABb2wGJQACKKBFBxcGBQEMC8IHpcicBwNiKGxlIy8kqqugD8UAAUAEwA7AAsAJwAlBzcfN3CBuKSANoARE3mqvMAumpAA)
The new error seems "good enough"?

**Comment by jakebailey** [comment]
Just looking at old issues; I definitely agree with the above. This was fixed by #45354 in the same was as #45333.

And as Daniel said in that PR:

> I think we'd really like to suggest `typeof Color.Red.toString` first if that would be valid, then fall back to using indexed access syntax.
